pkg/cli/clienturl/client_url.go

```diff
@@ -274,3 +277,27 @@ func MakeURLParserFn(
 		return clientOpts.ExplicitURL, nil
 	}
 }
+
+// checkMistakes reports likely mistakes to the user via warning messages.
+func (u *urlParser) checkMistakes(purl *pgurl.URL) {
+	if u.warnFn == nil {
+		return
+	}
+
+	// Check mistaken use of -c as direct option, instead of using
+	// options=-c...
+	opts := purl.GetExtraOptions()
+	for optName := range opts {
+		if strings.HasPrefix(optName, "-c") {
+			u.warnFn("\nwarning: found raw URL parameter \"%[1]s\"; "+
+				"are you sure you did not mean to use \"options=%[1]s\" instead?\n\n", optName)
+		}
```

Comment:
If you wanted, some of the surrounding formatting could go into the warnFn to keep it consistent as we add more checks.  But I don't feel strongly about it.